book/sources/exex/remote/src/exex.rs

```diff
@@ -0,0 +1,88 @@
+use futures_util::StreamExt;
+use remote_exex::proto::{
+    self,
+    remote_ex_ex_server::{RemoteExEx, RemoteExExServer},
+};
+use reth_exex::{ExExContext, ExExEvent, ExExNotification};
+use reth_node_api::FullNodeComponents;
+use reth_node_ethereum::EthereumNode;
+use reth_tracing::tracing::info;
+use std::sync::Arc;
+use tokio::sync::{broadcast, mpsc};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{transport::Server, Request, Response, Status};
+
+struct ExExService {
+    notifications: Arc<broadcast::Sender<ExExNotification>>,
+}
+
+#[tonic::async_trait]
+impl RemoteExEx for ExExService {
+    type SubscribeStream = ReceiverStream<Result<proto::ExExNotification, Status>>;
+
+    async fn subscribe(
+        &self,
+        _request: Request<proto::SubscribeRequest>,
+    ) -> Result<Response<Self::SubscribeStream>, Status> {
+        let (tx, rx) = mpsc::channel(1);
+
+        let mut notifications = self.notifications.subscribe();
+        tokio::spawn(async move {
+            while let Ok(notification) = notifications.recv().await {
+                let proto_notification = proto::ExExNotification {
+                    data: bincode::serialize(&notification).expect("failed to serialize"),
+                };
+                tx.send(Ok(proto_notification))
+                    .await
+                    .expect("failed to send notification to client");
+
+                info!("Notification sent to the gRPC client");
+            }
+        });
+
+        Ok(Response::new(ReceiverStream::new(rx)))
+    }
+}
+
+async fn remote_exex<Node: FullNodeComponents>(
+    mut ctx: ExExContext<Node>,
+    notifications: Arc<broadcast::Sender<ExExNotification>>,
+) -> eyre::Result<()> {
+    while let Some(result) = ctx.notifications.next().await {
+        let notification = result?;
```

Comment:
and in other places too
```suggestion
    while let Some(result) = ctx.notifications.try_next().await? {
```

Reply:
Done